test/integration/model.test.js

```diff
@@ -1665,6 +1666,54 @@ describe(Support.getTestDialectTeaser('Model'), function() {
       });
     });
 
+    describe("options sent to aggregate", function () {
+      var options, aggregateSpy;
+
+      beforeEach(function () {
+        options = { where: ['username = ?', 'user1']};
+
+        aggregateSpy = sinon.spy(this.User, "aggregate");
+      });
+
+      afterEach(function () {
+        var optsArg = aggregateSpy.args[0][2];
+
+        expect(optsArg.where).to.deep.equal(['username = ?', 'user1']);
+
+        aggregateSpy.restore();
+      });
+
+      it('modifies option "limit" by setting it to null', function() {
+        options.limit = 5;
+
+        return this.User.count(options).then(function() {
+          var optsArg = aggregateSpy.args[0][2];
+
+          expect(optsArg.limit).to.equal(null);
+        });
+      });
+
+      it('modifies option "offset" by setting it to null', function() {
+        options.offset = 10;
+
+        return this.User.count(options).then(function() {
+          var optsArg = aggregateSpy.args[0][2];
+
+          expect(optsArg.offset).to.equal(null);
+        });
+      });
+
+      it('modifies option "order" by setting it to null', function() {
+        options.order = "username";
+
+        return this.User.count(options).then(function() {
+          var optsArg = aggregateSpy.args[0][2];
+
+          expect(optsArg.order).to.equal(null);
```

Comment:
`expect(aggregateSpy).to.have.been.calledWith(sinon.match.any, sinon.match.any, sinon.match.object.and.has('order', null))`

Might not exactly be prettier, but I think it convey the assertion better, and it doesn't poke directly into the args array